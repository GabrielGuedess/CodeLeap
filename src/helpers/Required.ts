export type Required<T> = {
  [P in keyof T]-?: T[P];
};
