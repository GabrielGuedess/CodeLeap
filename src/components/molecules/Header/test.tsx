import { fireEvent, screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import { Header } from '.';

// eslint-disable-next-line @typescript-eslint/no-var-requires, unicorn/prefer-module
const useRouter = jest.spyOn(require('next/router'), 'useRouter');

const pushMock = jest.fn();
const signOutMock = jest.fn();

useRouter.mockImplementation(() => ({
  push: pushMock,
}));

jest.mock('hooks/useAuth', () => ({
  useAuth: () => ({
    signOut: signOutMock,
  }),
}));

describe('<Header />', () => {
  it('should render the header correctly', () => {
    // Arrange
    const { container } = renderWithTheme(<Header title="Title" />);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should change page when click on sign-out', () => {
    renderWithTheme(<Header title="Title" />);

    // Act
    fireEvent.click(screen.getByLabelText('sign-out'));

    // Assert
    expect(pushMock).toHaveBeenCalled();
    expect(signOutMock).toHaveBeenCalled();
  });
});
