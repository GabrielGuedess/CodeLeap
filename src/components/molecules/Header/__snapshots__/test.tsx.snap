// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the header correctly 1`] = `
.c1 {
  font-size: 2.2rem;
  font-weight: 700;
  color: #FFFFFF;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 2.7rem 3.7rem;
  color: #FFFFFF;
  background: #6084e9;
}

.c0 svg {
  cursor: pointer;
}

<header
  class="c0"
>
  <h1
    class="c1"
    color="secondary"
  >
    Title
  </h1>
  <svg
    aria-label="sign-out"
    fill="currentColor"
    height="32"
    viewBox="0 0 256 256"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="256"
      width="256"
    />
    <polyline
      fill="none"
      points="174 86 216 128 174 170"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
    />
    <line
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="104"
      x2="216"
      y1="128"
      y2="128"
    />
    <path
      d="M104,216H48a8,8,0,0,1-8-8V48a8,8,0,0,1,8-8h56"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
    />
  </svg>
</header>
`;
